pkg/storage/s3/storage.go

```diff
@@ -76,43 +75,17 @@ func (s Storage) BaseURL() *url.URL {
 
 // Save implements the (github.com/gomods/athens/pkg/storage).Saver interface.
 func (s *Storage) Save(ctx context.Context, module, version string, mod []byte, zip io.Reader, info []byte) error {
-	errChan := make(chan error, 3)
-
-	tctx, cancel := context.WithTimeout(ctx, env.Timeout())
-	defer cancel()
-
-	go s.upload(tctx, errChan, module, version, "mod", bytes.NewReader(mod))
-	go s.upload(tctx, errChan, module, version, "zip", zip)
-	go s.upload(tctx, errChan, module, version, "info", bytes.NewReader(info))
-
-	var errors error
-	for i := 0; i < 3; i++ {
-		err := <-errChan
-		if err != nil {
-			errors = multierror.Append(errors, err)
-		}
-	}
-	close(errChan)
-
-	return errors
+	err := modupl.Upload(ctx, module, version, bytes.NewReader(info), bytes.NewReader(mod), zip, s.upload)
+	return err
```

Comment:
Here you can just return the function call.

Reply:
I wanted to leave it here because at some point we  have to check the error and update the list file. I'll write a comment here.